receiver/awscontainerinsightreceiver/internal/host/nodeCapacity.go

```diff
@@ -0,0 +1,100 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/mem"
+	"go.uber.org/zap"
+)
+
+const (
+	goPSUtilProcDirEnv = "HOST_PROC"
+)
+
+type NodeCapacityProvider interface {
+	GetMemoryCapacity() int64
+	GetNumCores() int64
+}
+
+type NodeCapacity struct {
+	MemCapacity int64
+	CPUCapacity int64
+	logger      *zap.Logger
+
+	// osLstat returns a FileInfo describing the named file.
+	osLstat func(name string) (os.FileInfo, error)
+	// osSetenv sets the value of the environment variable named by the key
+	osSetenv      func(key string, value string) error
+	virtualMemory func() (*mem.VirtualMemoryStat, error)
+	cpuInfo       func() ([]cpu.InfoStat, error)
+}
+
+type nodeCapacityOption func(*NodeCapacity)
+
+func NewNodeCapacity(logger *zap.Logger, options ...nodeCapacityOption) (NodeCapacityProvider, error) {
+	nc := &NodeCapacity{
+		logger:        logger,
+		osLstat:       os.Lstat,
+		osSetenv:      os.Setenv,
+		virtualMemory: mem.VirtualMemory,
+		cpuInfo:       cpu.Info,
+	}
+
+	for _, opt := range options {
+		opt(nc)
+	}
+
+	if _, err := nc.osLstat("/rootfs/proc"); os.IsNotExist(err) {
```

Comment:
define `/rootfs/proc` as common VAR, saw it is used in a few places.

Reply:
I think it's better to define `/rootfs` and the concat it with `/proc` as we mount host `/` to `/rootfs` based on [cloudwatch agent](https://github.com/aws/amazon-cloudwatch-agent/blob/5be59ed097255f85cde797ef15df019d1fc9dcd4/amazon-cloudwatch-container-insights/k8s-deployment-manifest-templates/deployment-mode/daemonset/combination/combination.yaml#L174-L191)

Reply:
Sure. I will define some common constants.